[ALL TESTS] Update (#5861)
mlflow/mlflow: CHANGELOG.md

# CHANGELOG

> ## 1.25.2.dev0 (2022-05-13)

Reviewer: ```suggestion
## 1.26.0 (2022-05-13)
```

> MLflow 1.25.2.dev0 includes several major features and improvements:

Reviewer 2: ```suggestion
MLflow 1.26.0 includes several major features and improvements:
```

Features:

> - Add endpoint changes to the deployment CLI. (#5731, @trangevi)

Reviewer 2: ```suggestion
- Add endpoint naming and options configuration to the deployment CLI. (#5731, @trangevi)
```

> - Dev env setup script (#5717, @BenWilson2)

Reviewer 3: Add development environment setup script

Reviewer 2: Add development environment setup script for Linux and MacOS x86 Operating Systems

> - Dev env setup script (#5717, @BenWilson2)

Reviewer 2: ```suggestion
- [Development] Add development environment setup script for Linux and MacOS x86 Operating Systems (#5717, @BenWilson2)
```

> - Update mlflow.set_tracking_uri to support pathlib.Path (#5820) (#5824, @cacharle)

Reviewer 2: ```suggestion
- Update `mlflow.set_tracking_uri` to add support for paths defined as `pathlib.Path` in addition to existing `str` path declarations (#5820, #5824 @cacharle)
```

> - Scoring server timeout #5381 (#5663, @sniafas)

Reviewer 3: ```suggestion
Add timeout option for starting scoring server command
```

> - Scoring server timeout #5381 (#5663, @sniafas)

Reviewer 2: ```suggestion
- Add custom timeout override option to the scoring server CLI to support high latency models (#5663, @sniafas)
```

> - [FIX-5789] Added sticky header to experiment run list (#5818, @hubertzub-db)

Reviewer 3: ```suggestion
- [FIX-5789] Add sticky header to experiment run list (#5818, @hubertzub-db)
```

> - Added mechanism to delete artifacts from GCS repository using MLFlow garbage collection (#5811, @aditya-iyengar-rtl-de)

Reviewer 3: ```suggestion
- Add mechanism to delete artifacts from GCS repository using MLFlow garbage collection (#5811, @aditya-iyengar-rtl-de)
```

> - Added mechanism to delete artifacts from GCS repository using MLFlow garbage collection (#5811, @aditya-iyengar-rtl-de)

Reviewer 2: ```suggestion
- [Artifacts] Add GCS support for MLflow garbage collection (`mlflow gc`) (#5811, @aditya-iyengar-rtl-de)
```

> - Pass in the `pos_label` parameter to `eval_and_log_metrics` (#5807, @yxiong)

Reviewer 3: ```suggestion
- Add `pos_label` argument for `eval_and_log_metrics` API (#5807, @yxiong)
```

> - Pass in the `pos_label` parameter to `eval_and_log_metrics` (#5807, @yxiong)

Reviewer 2: ```suggestion
- [Evaluate] Add `pos_label` argument for `eval_and_log_metrics` API to support accurate binary classifier evaluation metrics (#5807, @yxiong)
```

> - Show latest, minimum and maximum metric values on metric page (#5574, @adamreeve)

Reviewer 2: ```suggestion
- [UI] Add fields for latest, minimum and maximum metric values on metric display page (#5574, @adamreeve)
```

- Autologging of input example and signature for pyspark ml (#5719, @bali0019)

> - Add virtualenv support for `mlflow models docker-build` (#5728, @harupy)

Reviewer 2: ```suggestion
- [Models] Add virtualenv environment manager support for `mlflow models docker-build` CLI (#5728, @harupy)
```

> - feat: support using wildcards in log_model_allowlist (#5723, @serena-ruan)

Reviewer 3: ```suggestion
- Support using wildcards in log_model_allowlist (#5723, @serena-ruan)
```

> - feat: support using wildcards in log_model_allowlist (#5723, @serena-ruan)

Reviewer 2: ```suggestion
- [Models] Add support for wildcard module matching in log_model_allowlist for PySpark models (#5723, @serena-ruan)
```

> - Add virtualenv support for MLflow projects (#5631, @harupy)

Reviewer 2: ```suggestion
- [Projects] Add `virtualenv` environment manager support for MLflow projects (#5631, @harupy)
```

> - Document virtualenv suppport for MLflow projects (#5727, @harupy)

Reviewer 3: Move to documentation updates section

> - Document virtualenv suppport for MLflow projects (#5727, @harupy)

Reviewer 2: ```suggestion
- [Docs] Add documentation for `virtualenv` environment manager support for MLflow projects (#5727, @harupy)
```
(and move to docs)

> - Make `pyfunc.spark_udf` support virtualenv (#5676, @WeichenXu123)

Reviewer 3: Add virtualenv support for `mlflow.pyfunc.spark_udf` API

> - Make `pyfunc.spark_udf` support virtualenv (#5676, @WeichenXu123)

Reviewer 2: ```suggestion
- [Models] Add `virtualenv` support for `mlflow.pyfunc.spark_udf` API (#5676, @WeichenXu123)
```

> - Added autologging of input example and signature for tf.estimator and keras fit_generator (#5510, @bali0019)

Reviewer 3: ```suggestion
- Add autologging of input example and signature for tf.estimator and keras fit_generator (#5510, @bali0019)
```

> - Adds JSON Schema Validation (#5458, @mrkaye97)

Reviewer 3: ```suggestion
- Add JSON schema validation (#5458, @mrkaye97)
```

> - Adds JSON Schema Validation (#5458, @mrkaye97)

Reviewer 2: ```suggestion
- [Server-infra] Add JSON Schema Type Validation to enable raising 400 errors on malformed requests to REST API endpoints (#5458, @mrkaye97)
```

> - Update UI (#5711, @harupy)

Reviewer 3: @harupy How to refine the description of the commit ?

Reviewer: #5711 didn't add new features or fix bugs. We can move this the small bug fixes and doc updates section.

> - Introducing abstract endpoint methods (#5378, @trangevi)

Reviewer 3: ```suggestion
- Introduce abstract `endpoint` interface for mlflow deployments (#5378, @trangevi)
```

> - Adding end time and run duration to Compare Runs view (#3378, @RealArpanBhattacharya)

Reviewer 3: ```suggestion
- Add end time and run duration to Compare Runs view (#3378, @RealArpanBhattacharya)
```

> - Adding end time and run duration to Compare Runs view (#3378, @RealArpanBhattacharya)

Reviewer 2: ```suggestion
- [UI] Add 'end time' and 'run duration' fields to Compare Runs view (#3378, @RealArpanBhattacharya)
```

> - Use schema when parsing csv (#5531, @vvijay-bolt)

Reviewer 3: ```suggestion
- Validate schema when parsing csv data during model serving (#5531, @vvijay-bolt)
```

> - Use schema when parsing csv (#5531, @vvijay-bolt)

Reviewer 2: ```suggestion
- [Serving] Add schema validation support when parsing input csv data for model serving (#5531, @vvijay-bolt)
```

Bug fixes and documentation updates:

> - Spark Datasource autologging: Fix REPL ID propagation from datasource listener to publisher (#5826, @dbczumar)

Reviewer 2: ```suggestion
- [Models] Fix REPL ID propagation from datasource listener to publisher for Spark data sources (#5826, @dbczumar)
```

> - Update ag-grid and implement getRowId to improve runs table performance (#5725, @adamreeve)

Reviewer 2: ```suggestion
- [UI] Update `ag-grid` and implement `getRowId` to improve performance in the runs table visualization (#5725, @adamreeve)
```

> - Fix tf-serving parsing when input are columnar formatter (#5825, @arjundc-db)

Reviewer 2: ```suggestion
- [Serving] Fix `tf-serving` parsing to support columnar-based formatting (#5825, @arjundc-db)
```

> - Fix log_artifact for large model in HDFS (#5812, @hitchhicker)

Reviewer 2: ```suggestion
- [Artifacts] Update `log_artifact` to support models larger than 2GB in HDFS (#5812, @hitchhicker)
```

> - fix metric names with "@" signs for lightgbm autolog (#5785, @mengchendd)

Reviewer 3: ```suggestion
- fix logging metric failure on metric names with "@" signs for lightgbm autolog (#5785, @mengchendd)
```

> - fix metric names with "@" signs for lightgbm autolog (#5785, @mengchendd)

Reviewer 2: ```suggestion
- [Models] Fix autologging to support `lightgbm` metric names with "@" symbols within their names (#5785, @mengchendd)
```

> - Pyfunc: Fix code dir resolution of subdirectories (#5806, @dbczumar)

Reviewer 3: ```suggestion
- Pyfunc: Fix code directory resolution of subdirectories (#5806, @dbczumar)
```

Reviewer 2: add [Models] to start of line

> - fix: fix mlflow server starts in R on windows (#5767, @serena-ruan)

Reviewer 3: ```suggestion
- Fix mlflow-R server starting failure on windows (#5767, @serena-ruan)
```

Reviewer 2: Add [Server-Infra] to start of line

> - Preview images with full width (#5606, @szczeles)

Reviewer 3: ```suggestion
- Preview images with full width for artifacts display (#5606, @szczeles)
```

> - Preview images with full width (#5606, @szczeles)

Reviewer 2: ```suggestion
- [UI] Fix artifacts display sizing to support full width rendering in preview pane (#5606, @szczeles)
```

- When loading spark model, bind spark.driver.bindAddress to 127.0.0.1 (#5753, @WeichenXu123)

> - Added autologging safety validation exemptions, and fixed autolog does not log implicit batch_size for tf keras fit method (#5683, @MarkYHZhang)

Reviewer 3: ```suggestion
- Add autologging safety validation exemptions, and fix the issue that autologging does not log implicit batch_size for `tensorflow.keras` model's `fit` method (#5683, @MarkYHZhang)
```

> - Adds JSON Schema Validation (#5458, @mrkaye97)

Reviewer 3: remove this line.

> - #4299 log_batch actually writes in batches (#5460, @erensahin)

Reviewer 3: ```suggestion
- Fix `SqlAlchemyStore.log_batch` implementation to make it log data in batches (#5460, @erensahin)
```

Reviewer 2: Add [Artifacts] to start of line

> - Scoring server timeout #5381 (#5663, @sniafas)

Reviewer 3: remove this line
